lib/js_defer.js

```diff
@@ -1613,58 +1452,64 @@ deferJsNs.DeferJs.prototype.noDeferCreateElementOverride = function() {
         }
       };
       deferJsNs.addOnload(elem, onload);
-      pagespeedutils.addHandler(elem, 'error', onload);
+      fasterizeutils.addHandler(elem, 'error', onload);
     }
     return elem;
   };
 };
 
-/**
- * @return {boolean} true if deferJs is running with experimental flag.
- */
-deferJsNs.DeferJs.prototype.isExperimentalMode = function() {
-  return deferJsNs.DeferJs.isExperimentalMode;
-};
-deferJsNs.DeferJs.prototype['isExperimentalMode'] =
-    deferJsNs.DeferJs.prototype.isExperimentalMode;
+deferJsNs.DeferJs.prototype.deferInsertScriptTag = function() {
+  var me = this;
+  Element.prototype.insertBefore = function (newElement, refElement) {
+    var elt = this;
+    if (newElement.nodeName === 'SCRIPT' && newElement.getAttribute('src')) {
+      if (me.state_ < deferJsNs.DeferJs.STATES.SYNC_SCRIPTS_DONE) {
+        me.log('Defer insertion of ' + newElement.getAttribute('src'));
+        me.submitTask(function() {
+          me.log('Dynamically insert: ' + newElement.getAttribute('src'));
+          me.originInsertBefore_.call(elt, newElement, refElement);
+          me.runNext();
+        });
+      } else {
+        me.log('Dynamically insert: ' + newElement.getAttribute('src'));
```

Comment:
It could be better to distinguish this case from the normal defered insertion in the log.